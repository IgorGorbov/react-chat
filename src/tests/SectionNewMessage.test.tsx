import * as React from "react";
import toJSON from 'enzyme-to-json';
import { mount } from "enzyme";
import SectionUserNewMessage from "../containers/SectionNewMessage";
import {
  SectionNewMessage,
  UserInput,
  UserButton
} from '../StyledComponents'

describe('testing SectionUserNewMessage', () => {
  const wrapper = mount(<SectionUserNewMessage />);

  it("should contain SectionMessagesList", () => {
    expect(wrapper.find(SectionNewMessage)).toHaveLength(1);
  });

  it("should contain UserInput", () => {
    expect(wrapper.find(UserInput)).toHaveLength(1);
  });

  it("should contain UserButton", () => {
    expect(wrapper.find(UserButton)).toHaveLength(1);
  });
  
  it('snapshot SectionUserNewMessage', () => {
    const tree = toJSON(wrapper);
    expect(tree).toMatchSnapshot();
  })
});
