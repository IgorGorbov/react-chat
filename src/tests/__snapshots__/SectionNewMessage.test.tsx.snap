// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing SectionUserNewMessage snapshot SectionUserNewMessage 1`] = `
<SectionUserNewMessage>
  <styled.section>
    <section
      className="sc-iwsKbI ioDznA"
    >
      <Styled(Input)
        inputwidth="92%"
        placeholder="Write your message ..."
      >
        <Input
          className="sc-ifAKCX jpARSZ"
          inputwidth="92%"
          placeholder="Write your message ..."
          type="text"
        >
          <div
            className="ui input sc-ifAKCX jpARSZ"
            inputwidth="92%"
          >
            <input
              onChange={[Function]}
              placeholder="Write your message ..."
              type="text"
            />
          </div>
        </Input>
      </Styled(Input)>
      <Styled(Button)
        buttonheight="40px"
        buttonmargin="0 -4px 0 0"
        buttonwidth="8%"
        icon={true}
        primary={true}
        size="large"
      >
        <Button
          as="button"
          buttonheight="40px"
          buttonmargin="0 -4px 0 0"
          buttonwidth="8%"
          className="sc-htoDjs hCMBMs"
          icon={true}
          primary={true}
          role="button"
          size="large"
        >
          <button
            buttonheight="40px"
            buttonmargin="0 -4px 0 0"
            buttonwidth="8%"
            className="ui large icon primary button sc-htoDjs hCMBMs"
            onClick={[Function]}
            role="button"
          >
            <Icon
              as="i"
              name="send"
            >
              <i
                aria-hidden="true"
                className="send icon"
              />
            </Icon>
          </button>
        </Button>
      </Styled(Button)>
    </section>
  </styled.section>
</SectionUserNewMessage>
`;
